README.md

```diff
@@ -105,17 +106,16 @@ reference.
 ### Serializing to the wire
 
 ```php
-    use OpenTracing\Carriers\HttpHeaders as HttpHeadersCarrier;
-    use OpenTracing\Context;
     use OpenTracing\GlobalTracer;
+    use OpenTracing\Formats;
     
     ...
     
     $tracer = GlobalTracer::get(); 
     
     $spanContext = $tracer->extract(
-        Propagator::HTTP_HEADERS,
-        HttpHeaders::withHeaders($request->getHeaders())
+        Formats\HTTP_HEADERS,
+        $request
```

Comment:
how is the tracer implementation supposed to know how to operate on the `request`? Different frameworks have different notions of the request, the tracer cannot be responsible for knowing how to access each of them. The Carrier APIs like TextMapReader provide the necessary level of abstraction by giving the tracer a stable API to work against, and the carrier translates that API into the operations on the underlying `request` object (or more typically, request headers).

Reply:
Fair point @yurishkuro I just updated the documentation, please have a look.

Regarding the TextMapReader I don't think we need that anymore. The `array` and the `ArrayObject` types are enough for this purposes.